course/viewbase.py

```diff
@@ -57,31 +57,36 @@ def get_resource_objects(self):
             self.instance = instance
             self.course = self.instance.course
             self.content = CachedContent(self.instance)
+            self.user_in_course = None
+            if user.is_authenticated:
+                self.user_in_course = self.instance.get_enrollment_for(user)
             self.is_student = self.instance.is_student(user)
             self.is_assistant = self.instance.is_assistant(user)
             self.is_teacher = self.course.is_teacher(user)
             self.is_course_staff = self.is_teacher or self.is_assistant
             self.get_taggings = lambda: CachedStudent(instance, user.id).data['tag_slugs']
 
             self.note(
-                "course", "instance", "content", "is_student", "is_assistant",
+                "course", "instance", "content", "user_in_course", "is_student", "is_assistant",
                 "is_teacher", "is_course_staff", "get_taggings",
             )
 
             # Apply course instance language.
             if self.instance.language:
-                lang = self.instance.language
-                if lang.startswith("|"):
-                    active = get_language()
-                    if "-" in active:
-                        active = active.split("-")[0]
-                    if "|" + active + "|" in lang:
-                        translation.activate(active)
-                    else:
-                        fallback = lang[1:lang.find("|", 1)]
-                        translation.activate(fallback)
-                else:
-                    translation.activate(lang)
+                instance_languages = self.instance.language
+
+                languages = []
+                if self.user_in_course and self.user_in_course.language:
+                    languages.append(self.user_in_course.language)
+                if user.is_authenticated and user.userprofile.language:
+                    languages.append(user.userprofile.language)
+                languages.append(get_language())
+                language = languages[0].split("-")[0]
```

Comment:
I think this would benefit from a loop, i.e., it would try to find a language that is defined for the course. If not, then the above list could be replaced with just a `if .. elif .. else`.

Reply:
Thanks, I'll try to make it use a loop. 

Is it needed to check that the language is defined for the course? So for example if the course is only available in English, should the course page be in English when user has set their profile language to be Finnish? If the course is not available in the given language, it only affects the side bar and dates etc but not the actual content, like in this pic:

![518-course-language](https://user-images.githubusercontent.com/36920208/88660965-612fdd80-d0e0-11ea-8473-d7ebdb887947.png)

Reply:
Idea is that all content in the page is in the same language, hence we try to find a language that is defined for the course.

Reply:
Ok, that clears it :+1: